File: mobile/src/main/java/com/dailymeditation/android/activities/SplashActivity.java
package com.dailymeditation.android.activities;

import android.content.Intent;
import android.graphics.PorterDuff;
import android.graphics.PorterDuffColorFilter;
import android.graphics.drawable.Drawable;
import android.os.Bundle;
import android.os.Handler;
import android.view.View;
import android.widget.ImageView;

import androidx.appcompat.app.AppCompatActivity;

import com.dailymeditation.android.R;

public class SplashActivity extends AppCompatActivity {

    private static final int SPLASH_DISPLAY_LENGTH = 1000;

    ImageView mSplashIcon;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);
        mSplashIcon = findViewById(R.id.splash_icon);
        init();
        gotoNextScreen();
    }

    private void init() {
        if (getSupportActionBar() != null) {
            getSupportActionBar().hide();
        }
        Drawable mDrawable = getResources().getDrawable(R.mipmap.ic_launcher);
        mDrawable.setColorFilter(new PorterDuffColorFilter(getResources().getColor(R.color.splash_icon_color), PorterDuff.Mode.MULTIPLY));
        mSplashIcon.setImageDrawable(mDrawable);

        getWindow().getDecorView().setSystemUiVisibility(View.SYSTEM_UI_FLAG_FULLSCREEN);
    }

    private void gotoNextScreen() {
        new Handler().postDelayed(() -> {
            startActivity(new Intent(SplashActivity.this, MainActivity.class));
            finish();
        }, SPLASH_DISPLAY_LENGTH);
    }
}

File: mobile/src/main/java/com/dailymeditation/android/activities/SplashActivity.kt
package com.dailymeditation.android.activities

import android.content.Intent
import android.graphics.PorterDuff
import android.graphics.PorterDuffColorFilter
import android.os.Bundle
import android.os.Handler
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.res.ResourcesCompat
import com.dailymeditation.android.R
import kotlinx.android.synthetic.main.activity_splash.*

class SplashActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)
        init()
        gotoNextScreen()
    }

    private fun init() {
        supportActionBar?.hide()
        ResourcesCompat.getDrawable(resources, R.mipmap.ic_launcher, null)?.let {
            it.colorFilter = PorterDuffColorFilter(
                ResourcesCompat.getColor(resources, R.color.splash_icon_color, null),
                PorterDuff.Mode.MULTIPLY
            )
            splash_icon?.setImageDrawable(it)
        }
        window.decorView.systemUiVisibility = View.SYSTEM_UI_FLAG_FULLSCREEN
    }

    private fun gotoNextScreen() {
        Handler().postDelayed({
            startActivity(Intent(this@SplashActivity, MainActivity::class.java))
            finish()
        }, SPLASH_DISPLAY_LENGTH.toLong())
    }

    companion object {
        private const val SPLASH_DISPLAY_LENGTH = 1000
    }
}